src/main/java/org/candlepin/subscriptions/capacity/CapacityReconciliationController.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright Red Hat, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Red Hat trademarks are not licensed under GPLv3. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.subscriptions.capacity;
+
+import static org.candlepin.subscriptions.db.model.SubscriptionCapacity.from;
+
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.MeterRegistry;
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
+import org.candlepin.subscriptions.capacity.files.ProductWhitelist;
+import org.candlepin.subscriptions.db.OfferingRepository;
+import org.candlepin.subscriptions.db.SubscriptionCapacityRepository;
+import org.candlepin.subscriptions.db.model.*;
```

Comment:
I'm not sure if the swatch team has a standard on star imports, but for teamnado at least, it is not allowed. You may want to bump up the default IntelliJ value from 4 to 99 or something like that.

Reply:
I am curious as to the purpose of that rule, but I have added the specific imports anyway.

Reply:
I don't know if there is such a rule or not for this project. But for explanations on why star imports should be avoided, there is [this stackoverflow answer](https://stackoverflow.com/a/147461) and a blog post "[import on demand is evil](http://www.javadude.com/posts/20040522-import-on-demand-is-evil/)". Admittedly the "no star imports" rule isn't without controversy, as the [second-most-popular answer](https://stackoverflow.com/a/149282) from the same stackoverflow question above says that you should prefer star imports instead.

The main purpose of the rule is so that if one updates a dependency, and that library happens to add a class whose name is already star imported elsewhere, then the project has a compilation failure. I don't know how often it happens, but apparently it does.

Ultimately I think this is ultimately a team-standard thing. For this project, star imports appear to be fine: They are used quite a bit in the test classes, and all of the Roller classes are doing a static star import for Granularity.

Reply:
We have generally been against * imports ... except for in unit tests for classes such as Mockito and Jupiter/Junit assertions. I personally wouldn't have even noticed though :smile:

Reply:
Removed the * imports.

Reply:
Removed the wildcard imports.